Mathlib/Data/Polynomial/Degree/Definitions.lean

```diff
@@ -1180,6 +1194,9 @@ theorem eq_C_of_natDegree_eq_zero (h : natDegree p = 0) : p = C (coeff p 0) :=
   eq_C_of_natDegree_le_zero h.le
 #align polynomial.eq_C_of_nat_degree_eq_zero Polynomial.eq_C_of_natDegree_eq_zero
 
+lemma natDegree_eq_zero {p : R[X]} : p.natDegree = 0 ↔ ∃ x, C x = p :=
```

Comment:
This is a less precise version of [Polynomial.eq_C_coeff_zero_iff_natDegree_eq_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Degree/Definitions.html#Polynomial.eq_C_coeff_zero_iff_natDegree_eq_zero) below, so I don't see the point of having it.

Reply:
These existentials are sometimes useful because then you can `subst` away the polynomial, whilst if you have it as `p.coeff 0`, you cannot

Reply:
On the other hand, this should certainly be proved in terms of that as opposed to a separate proof

Reply:
I see, fair enough!